serialport/bindings-cpp
# Bring everything up-to-date

Ended up refactoring quite a bit while investigating a bug (_seems prebuilds are failing on more and more setups_), figured a recap PR might help...

- Fix outdated packages (that don't require more refactoring)
- Update node version in `test` to match that of `prebuild` action
- Update napi_versions to 8
- Update prebuildify target to 20
- Update (split) matrix
  - force tags where missing
  - add `win32-arm64`
  - fix outdated image for linux-x64 (`centos7-devtoolset7` => `almalinux-devtoolset11`)
  - switch `linux-arm64-lts` to `linux-arm64`
- Update artifact names with `matrix.name` to avoid conflicts (merged on download)
- Fix lint in tests

TODO:

- **Currently, this removes the `semantic-release` step in favor of a `tree prebuilds` to finish debugging ([naming seems to have changed](https://www.npmjs.com/package/@serialport/bindings-cpp?activeTab=code): impact?), see action run:**
  - https://github.com/Nerivec/bindings-cpp/actions/runs/12331013585
- Can probably also add `dockcross/linux-riscv64`, which is becoming more requested? (although not yet in `prebuild/docker-images`)

_Feel free to update/split as desired._
